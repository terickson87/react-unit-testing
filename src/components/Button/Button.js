import React from 'react';
import './button.css';

function Button({label}) {

   return (
      <div data-testid="Button" className="button-style">{label}</div>
   )
}

export default Button;