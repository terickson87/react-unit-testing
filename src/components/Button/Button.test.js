import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom";
import ReactTestUtils from 'react-dom/test-utils';
import TestRenderer from 'react-test-renderer';

import Button from './Button';


afterEach(cleanup);

it("renders without crashing", () => {
   const div = document.createElement("div");
   ReactDOM.render(<Button />, div);
 });

 it("renders Button Correctly 1", () => {
   const {getByTestId} = render(<Button label="click me please"/>);
   expect(getByTestId("Button")).toHaveTextContent("click me please");
 });
 
 it("renders Button Correctly 2", () => {
   const {getByTestId} = render(<Button label="save"/>);
   expect(getByTestId("Button")).toHaveTextContent("save");
 });

it("Matches Snapshot 1", () => {
   const tree = TestRenderer.create(<Button label="save"/>).toJSON();
   expect(tree).toMatchSnapshot();
})

it("Matches Snapshot 2", () => {
  const tree = TestRenderer.create(<Button label="click me please"/>).toJSON();
  expect(tree).toMatchSnapshot();
})