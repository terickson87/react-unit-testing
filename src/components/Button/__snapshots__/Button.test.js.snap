// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1 1`] = `
<div
  className="button-style"
  data-testid="Button"
>
  save
</div>
`;

exports[`Matches Snapshot 2 1`] = `
<div
  className="button-style"
  data-testid="Button"
>
  click me please
</div>
`;
